.pre-commit-config.yaml

```diff
@@ -0,0 +1,16 @@
+repos:
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.8.1
+    hooks:
+      - id: ruff
+        name: Run ruff on tests/
+        args: [--exit-non-zero-on-fix]
+        files: ^tests/
```

Comment:
We should check all files, like current "test" action does.  BTW, ruff test in the test.yml now redundant and should be removed.